src/transformers/tokenization_utils_base.py

```diff
@@ -1454,12 +1459,20 @@ def __init__(self, **kwargs):
         model_max_length = kwargs.pop("model_max_length", kwargs.pop("max_len", None))
         self.model_max_length = model_max_length if model_max_length is not None else VERY_LARGE_INTEGER
 
-        # Padding side is right by default and overridden in subclasses. If specified in the kwargs, it is changed.
+        # Padding and truncation side are right by default and overridden in subclasses. If specified in the kwargs, it
+        # is changed.
         self.padding_side = kwargs.pop("padding_side", self.padding_side)
-        assert self.padding_side in [
-            "right",
-            "left",
-        ], f"Padding side should be selected between 'right' and 'left', current value: {self.padding_side}"
+        if self.padding_side not in ["right", "left"]:
+            raise ValueError(
+                f"Padding side should be selected between 'right' and 'left', current value: {self.padding_side}"
+            )
```

Comment:
@Narsil, if I understand correctly, does it seem clearer to you if I include the PR #15454 change directly here?

Reply:
Oh I don't think it makes that much of a difference (it's relatively small changes both ways), you already splitted them, so you can keep them splitted.

But my main point about the test applies to both PRs.